Lib/test/test_collections.py

```diff
@@ -70,6 +70,11 @@ def test_dict_copy(self):
         obj[123] = "abc"
         self._copy_test(obj)
 
+    def test_str_rmod(self):
+        arg = UserString("python")
+        template = "I love %s"
+        self.assertEqual(arg.__rmod__(template), "I love python")
+        self.assertIs(arg.__rmod__(type('Dummy', (), {})), NotImplemented)
```

Comment:
It would be more convincing if the test reproduced the end-to-end example from the bpo issue (https://bugs.python.org/file46854/userstringerror.py).

Reply:
Okey, done.